crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -440,7 +452,9 @@ impl MultiThreadedExecutor {
             #[cfg(feature = "trace")]
             let system_guard = system_span.enter();
             let res = std::panic::catch_unwind(AssertUnwindSafe(|| {
-                // SAFETY: access is compatible
+                // SAFETY:
+                // - Access is compatible.
```

Comment:
This safety comment isn't right, but it's also not worse than it was before this PR. I'm leaving this fix for the future, since safety invariants have a tendency of "bubbling up" through functions.

Reply:
We should either leave a TODO here to fix it, file an issue to ensure it eventually gets fixed, or fix it here.  I'm in favor of proactively fixing things instead of kicking the can down the line.

Reply:
I've added a TODO. This file has a few implicit invariants and I still need to figure out what they are so I can properly document them.

I don't think we need to hold this current fix back for an unrelated issue.